lib/term.js

```diff
@@ -1,42 +1,55 @@
 const rdf = require('@rdfjs/data-model')
 
-function term (value, type, datatype) {
-  if (datatype) {
-    datatype = rdf.namedNode(datatype.value || datatype.toString())
+function blankNode (value) {
+  if (value && typeof value !== 'string') {
+    throw new Error('unknown type')
   }
 
-  if (type) {
-    if (type === 'BlankNode') {
-      return rdf.blankNode(value && value.toString())
-    }
+  return rdf.blankNode(value && value.toString())
+}
+
+function literal (value, languageOrDatatype) {
+  if (typeof value !== 'string' && typeof value !== 'number') {
+    throw new Error('unknown type')
+  }
 
-    if (type === 'Literal') {
-      return rdf.literal(value.toString(), datatype)
-    }
+  languageOrDatatype = languageOrDatatype && (languageOrDatatype.value || languageOrDatatype.toString())
 
-    if (type === 'NamedNode') {
-      return rdf.namedNode(value.toString())
-    }
+  if (languageOrDatatype && languageOrDatatype.indexOf(':') !== -1) {
+    languageOrDatatype = rdf.namedNode(languageOrDatatype)
   }
 
-  if (!value) {
-    return undefined
+  return rdf.literal(value.toString(), languageOrDatatype)
+}
+
+function namedNode (value) {
+  if (typeof value !== 'string') {
+    throw new Error('unknown type')
   }
 
-  if (typeof value === 'object' && value.termType) {
+  return rdf.namedNode(value)
+}
+
+function term (value, type = 'Literal', languageOrDatatype) {
+  // it's already a RDF/JS Term
+  if (value && typeof value === 'object' && value.termType) {
     return value
   }
 
-  // TODO: library for JavaScript type -> xsd
+  if (type === 'BlankNode') {
+    return blankNode(value)
+  }
+
+  if (!value) {
+    return undefined
+  }
```

Comment:
Should you reverse these `if` locks to check for truthiness before returning `blankNode()`?

Reply:
For `blankNode()` `null` or `undefined` is allowed, that's why the order should be OK.